pkg/pillar/types/zedroutertypes.go

```diff
@@ -2592,6 +2567,37 @@ func (config *NetworkInstanceConfig) IsIPv6() bool {
 	return false
 }
 
+// WithUplinkProbing returns true if the network instance is eligible
+// for uplink probing (see pkg/pillar/uplinkprober).
+func (config *NetworkInstanceConfig) WithUplinkProbing() bool {
+	if config.Type != NetworkInstanceTypeLocal && config.Type != NetworkInstanceTypeCloud {
+		return false
+	}
+	return IsSharedPortLabel(config.Logicallabel)
+}
```

Comment:
IMHO a case statement would be more readable that this logical expression.

Reply:
Changed to switch statement.

Reply:
Yetus generated one issue - missing some case statements, which does not make much sense to me since there is the default branch.

Reply:
Yeah we should turn off that yetus check - I've seen it fire elsewhere when there is a default case.